### DISCUSSION TITLE
Is there a way to group the errors by rules?

### DISCUSSION THREAD

**Discussion by mrdulin** [eslint/eslint#18585] [Q&A] [answered]
```sh

### more errors ...

D:\workspace\company\project\src\routes\zombie-user-statistics\storeSlice.ts
  2:1  error  There should be no empty line within import group              import/order
  4:1  error  There should be at least one empty line between import groups  import/order
  6:1  error  There should be at least one empty line between import groups  import/order

D:\workspace\company\project\src\storeSlice.ts
   2:1   error    There should be at least one empty line between import groups  import/order
  72:67  warning  Forbidden non-null assertion                                   @typescript-eslint/no-non-null-assertion

✖ 3336 problems (2824 errors, 512 warnings)
  2807 errors and 13 warnings potentially fixable with the `--fix` option.
```

By default, the errors grouped by file, is there a way to group the errors by rules and only fix errors of specific rules?

Error report grouped by rules:

```sh
✖ 3336 problems (2824 errors, 512 warnings)
import/order 1000 errors
prettier/prettier 10 errors
@typescript-eslint/no-explicit-any 13 warnings
```

Fix errors of specific rules:

```sh
npx eslint --fix-rules ["import/order"]
```

**Comment by Rec0iL99** [comment]
Hi @mrdulin, to group the linting results based on the rules, I think writing a [custom formatter](https://eslint.org/docs/latest/extend/custom-formatters) will help you here.

There's currently no option to fix the problems reported based on rules; however, there's a CLI option to fix rules based on the rule type: [--fix-type](https://eslint.org/docs/latest/use/command-line-interface#--fix-type).

**Comment by Rel1cx** [comment, marked as answer]
https://github.com/mizdra/eslint-interactive
Use this. It's a great tool.

**Reply by mrdulin** [reply]
This is really what I want. Thx!
